tools/fuzzy_proof_generator_tool/src/FuzzyProofGenerator.ts

```diff
@@ -0,0 +1,464 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+
+import Util from './Util';
+import NodeBase from './NodeBase';
+import BranchNode from './BranchNode';
+import ExtensionNode from './ExtensionNode';
+import LeafNode from './LeafNode';
+
+import assert = require('assert');
+import ethUtil = require('ethereumjs-util');
+import rlp = require('rlp');
+import crypto = require('crypto');
+
+interface ProofData {
+  value: Buffer;
+  encodedPath: Buffer;
+  rlpParentNodes: Buffer;
+  root: Buffer;
+}
+
+const FuzzyProofGenerator = {
+
+  /** Generates a proof data. */
+  generateByPattern(
+    pattern: string,
+    pathMaxLength: number = 40,
+    valueMaxLength: number = 40,
```

Comment:
Shouldn't the default be 20 bytes in the context of ethereum?

Reply:
Actually 32. I would update it.

Reply:
20 in the case of accounts where we use the lib ;)